clients/src/main/java/org/apache/kafka/common/serialization/Serializer.java

```diff
@@ -34,20 +35,22 @@
      * @param configs configs in key/value pairs
      * @param isKey whether is for key or value
      */
-    public void configure(Map<String, ?> configs, boolean isKey);
+    void configure(Map<String, ?> configs, boolean isKey);
 
     /**
+     * Convert {@code data} into a byte array.
+     *
      * @param topic topic associated with data
      * @param data typed data
      * @return serialized bytes
      */
-    public byte[] serialize(String topic, T data);
+    byte[] serialize(String topic, T data);
 
     /**
      * Close this serializer.
      * This method has to be idempotent if the serializer is used in KafkaProducer because it might be called
```

Comment:
By the way, I wonder if we should just say it should be idempotent? Seems redundant to mention KafkaProducer.